LanguageFeatures/Records/record_type_annotations_A07_t02.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if a record type has any of:
+///
+/// The same field name more than once. This is true even if one or both of the
+/// colliding fields is positional. We could permit collisions with positional
+/// field names since they are only used for documentation, but we disallow it
+/// because it's confusing and not useful.
+///
+/// Only one positional field and no trailing comma. This isn't ambiguous, since
+/// there are no parenthesized type expressions in Dart. But prohibiting this is
+/// symmetric with record expressions and leaves the potential for later support
+/// for parentheses for grouping in type expressions.
+///
+/// A field named hashCode, runtimeType, noSuchMethod, or toString.
+///
+/// A field name that starts with an underscore.
+///
+/// A field name that collides with the synthesized getter name of a positional
+/// field. For example: (int, $0: int) since the named field '$0' collides with
+/// the getter for the first positional field.
+///
+/// @description Checks that it is no error if a record type has a field name
+/// that doesn't collide with the synthesized getter name of a positional field
+/// @author sgrekhov22@gmail.com
+/// @issue 498838
+
+// SharedOptions=--enable-experiment=records
+
+typedef R1 = (int $5, {String s});
```

Comment:
Similarly to the above, the compiler behavior for this test case is not yet defined. And same suggestion: either mark it as such or postpone adding this test until the language design decision is made.

/cc @lrhn